gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/ApplicationController.groovy

```diff
@@ -144,6 +150,20 @@ class ApplicationController {
     }
   }
 
+  @ApiOperation(value = "Generate a template from the pipeline configuration")
+  @RequestMapping(value = "/{application}/pipelineConfigs/{pipelineName:.+}/convertToTemplate", method = RequestMethod.GET)
+  String convertPipelineConfigToPipelineTemplate(
+    @PathVariable("application") String application, @PathVariable("pipelineName") String pipelineName) {
+    Map pipelineConfig = applicationService.getPipelineConfigsForApplication(application).find {
+      it.name == pipelineName
+    }
+    if (pipelineConfig == null) {
+      throw new NotFoundException("Pipeline config '${pipelineConfigId}' could not be found")
+    }
+    String template = orcaService.convertToPipelineTemplate(pipelineConfig).body.in().text
```

Comment:
We don't really want to have direct service calls in the controllers. Instead, this should get put into `ApplicationService` and wrapped with Hystrix.